src/zenml/zen_stores/migrations/versions/1cb6477f72d6_move_artifact_save_type.py

```diff
@@ -0,0 +1,83 @@
+"""move artifact save type [1cb6477f72d6].
+
+Revision ID: 1cb6477f72d6
+Revises: 1d8f30c54477
+Create Date: 2024-10-10 15:44:09.465210
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "1cb6477f72d6"
+down_revision = "1d8f30c54477"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    # Step 1: Add nullable save_type column to artifact_version
+    with op.batch_alter_table("artifact_version", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("save_type", sa.TEXT(), nullable=True))
+
+    # Step 2: Move data from step_run_output_artifact.type to artifact_version.save_type
+    op.execute("""
+        UPDATE artifact_version
+        SET save_type = step_run_output_artifact.type
+        FROM step_run_output_artifact
+        WHERE artifact_version.id = step_run_output_artifact.artifact_id
+    """)
+    op.execute("""
```

Comment:
Aren't the ones that have no entry in the `step_run_output_artifact` table external artifacts? At long as no step runs were deleted at least?

Reply:
Hm, yeah, sounds reasonable.